src/hubverse_transform/hub_config.py

```diff
@@ -46,6 +48,47 @@ def __repr__(self):
     def __str__(self):
         return f"Hubverse config information for {self.hub_name}."
 
+    def get_task_id_values(self, round_name: str = "all") -> dict[str, set]:
+        """
+        Return a dict of hub task ids and values for a specific round.
+
+        Parameters
+        ----------
+        round_name : str, default="all"
+            Which round's tasks to include. If "all", return tasks
+            and values for all rounds.
+
+        Returns
+        -------
+        model_tasks : dict[str, list]
+            A mapping of tasks ids to their possible values, as
+            defined in a hub's tasks.json configuration file.
+
+        Raises
+        ------
+        ValueError
+            If the tasks configuration file doesn't contain the
+            specified round_name.
+        """
+
+        tasks = self.tasks
+        rounds = tasks.get("rounds", [])
+        if round_name != "all":
+            rounds = [r for r in rounds if r.get("round_name") == round_name]
+
+        if len(rounds) == 0:
+            raise ValueError(f"Round {round_name} not found in tasks configuration")
+
+        model_tasks_dict: dict[str, set] = dict()
+
+        for r in rounds:
+            for task in r.get("model_tasks", []):
+                task_values = self._get_task_id_values(task)
+                for key, value in task_values.items():
+                    model_tasks_dict[key] = model_tasks_dict.get(key, set()) | value
```

Comment:
Is this line doing some appendage if `model_tasks_dict[key]` was already populated from a previous iteration of the loops over `r` or `task`?  Based on the unit test output below it seems like it does, but I don't know what `model_tasks_dict.get(key, set()) | value` does, would have expected to see something about `union` in there.

Reply:
Yep-- `|` is a shorthand syntax for doing a union!